src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/NonAssociationsService.kt

```diff
@@ -216,6 +236,18 @@ class NonAssociationsService(
     }
   }
 
+  private fun filterByPrisonId(nonAssociations: List<NonAssociationJPA>, prisonId: String): List<NonAssociationJPA> {
+    val prisonerNumbers = nonAssociations.map { nonna ->
+      listOf(nonna.firstPrisonerNumber, nonna.secondPrisonerNumber)
+    }.flatten()
```

Comment:
i'd use a set

Reply:
As discussed offline, I'm tweaking this to:
- use `flatMap()` to do collection/flattening in one go (e.g. removing the final `flatten()` call)
- make prisonerNumbers a `Set` just before sending the request to Offender Search API

PR with these tweaks coming soon